### ISSUE TITLE
SEATBELT_FROZEN doesn't fail the process when violations decrease

### ISSUE THREAD

**Issue by likewerealive** [justjake/eslint-seatbelt#18]
When using `SEATBELT_FROZEN=1`, I see the frozen violation message but ESLint still exits with code 0: `[eslint-seatbelt]: SEATBELT_FROZEN: Expected 3 errors, found 2.`

Looking at the source code, `messageFrozenUnderMaxErrorCount` returns severity 1 (warning) instead of severity 2 (error):

```javascript
function messageFrozenUnderMaxErrorCount(...) {
  return {
    ...message,
    severity: 1,  // ← Should be 2 to fail the process
    message: ...
  }
}
```

Since ESLint only exits non-zero for errors (not warnings), CI doesn't fail when the seatbelt file is out of date. Is it failing as a warning expected or should this be an error?

**Comment by woohyun-park** [comment]
I reproduced this behavior on the current `main` branch and prepared a focused fix that preserves error severity for frozen-mode decreases. I’ve opened https://github.com/justjake/eslint-seatbelt/pull/32 with a regression test.

If this overlaps with any work already in progress, I’d be happy to coordinate.
